artifacts/fedora/fedora.go

```diff
@@ -155,25 +161,35 @@ func (f *fedora) releaseMatches(release *Release) bool {
 
 func (f *fedoraGatherer) releaseMatches(release *Release) bool {
 	version, err := strconv.Atoi(release.Version)
-	return err == nil && version >= minimumVersion &&
-		release.Arch == f.Arch &&
-		release.Variant == f.Variant &&
-		strings.HasSuffix(release.Link, "qcow2")
+	if err != nil {
+		return false
+	}
+
+	for _, arch := range f.Arch {
+		if release.Version == f.Version && release.Arch == arch {
+			return version >= minimumVersion &&
+				release.Variant == f.Variant &&
+				strings.HasSuffix(release.Link, "qcow2")
+		}
+	}
+
+	return false
 }
 
-func New(release string, envVariables map[string]string) *fedora {
+func New(release, arch string, envVariables map[string]string) *fedora {
 	return &fedora{
 		Version:      release,
-		Arch:         "x86_64",
+		Arch:         arch,
 		Variant:      "Cloud",
 		getter:       &http.HTTPGetter{},
 		EnvVariables: envVariables,
 	}
 }
 
-func NewGatherer() *fedoraGatherer {
+func NewGatherer(version string, arch []string) *fedoraGatherer {
```

Comment:
Please allow configuration of env variables per arch, i.e. exchange `arch` with a struct that contains the architecture and the corresponding env variables.

Reply:
I removed arguments from `NewGatherer`. It looks like it was before now.

Reply:
Do you want to have a new argument of type struct that contains the architecture and environment variables?